blog/views.py

```diff
@@ -1,20 +1,28 @@
 from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
-from django.views.generic import CreateView, View, DetailView, DeleteView, UpdateView
+from django.views.generic import CreateView, DetailView, DeleteView, UpdateView, ListView
 from .forms import PostCreateForm, PostUpdateForm
 from django.urls import reverse_lazy
-from django.shortcuts import render
+from django.shortcuts import get_object_or_404
+from django.contrib.auth.models import User
 
 from .models import Post
+from user.models import Follow
 
 
-class HomeView(LoginRequiredMixin, View):
+class HomeView(LoginRequiredMixin, ListView):
+    model = User
+    template_name = "blog/home.html"
 
-    def get(self, request):
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
         post_list = Post.objects.all()
-        context = {
-            'post_list': post_list,
-        }
-        return render(request, 'blog/home.html', context)
+        login_user = self.request.user
+        following_count = Follow.objects.filter(follow_from=login_user).count()
+        follower_count = Follow.objects.filter(follow_to=login_user).count()
+        context['following_count'] = following_count
+        context['follower_count'] = follower_count
+        context['post_list'] = post_list
```

Comment:
このレベルであれば、わざわざ分けなくても大丈夫です
例
```suggestion
        context['following_count'] = Follow.objects.filter(follow_from=login_user).count()
```